packages/@aws-cdk/aws-glue/README.md

```diff
@@ -1,2 +1,187 @@
+    type: glue.Schema.array(Schema.string),
+    comment: 'col2 is an array of strings' // comment is optional
+  }]
+  dataFormat: glue.DataFormat.Json
+});
+```
+
+By default, a S3 bucket will be created to store the table's data but you can manually pass the `bucket` and `s3Prefix`:
+
+```ts
+new glue.Table(stack, 'MyTable', {
+  bucket: myBucket,
+  s3Prefix: 'my-table/'
+  ...
+});
+```
+
+#### Partitions
+
+To improve query performance, a table can specify `partitionKeys` on which data is stored and queried separately. For example, you might partition a table by `year` and `month` to optimize queries based on a time window:
+
+```ts
+new glue.Table(stack, 'MyTable', {
+  database: myDatabase,
+  tableName: 'my_table',
+  columns: [{
+    name: 'col1',
+    type: glue.Schema.string
+  }],
+  partitionKeys: [{
+    name: 'year',
+    type: glue.Schema.smallint
+  }, {
+    name: 'month',
+    type: glue.Schema.smallint
+  }],
+  dataFormat: glue.DataFormat.Json
+});
+```
+
+### [Encryption](https://docs.aws.amazon.com/athena/latest/ug/encryption.html)
+
+You can enable encryption on a Table's data:
+* `Unencrypted` - files are not encrypted. The default encryption setting.
+* [S3Managed](https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingServerSideEncryption.html) - Server side encryption (`SSE-S3`) with an Amazon S3-managed key.
+```ts
+new glue.Table(stack, 'MyTable', {
+  encryption: glue.TableEncryption.S3Managed
+  ...
+});
+```
+* [Kms](https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingKMSEncryption.html) - Server-side encryption (`SSE-KMS`) with an AWS KMS Key managed by the account owner.
+
+```ts
+// KMS key is created automatically
+new glue.Table(stack, 'MyTable', {
+  encryption: glue.TableEncryption.Kms
+  ...
+});
+
+// with an explicit KMS key
+new glue.Table(stack, 'MyTable', {
+  encryption: glue.TableEncryption.Kms,
+  encryptionKey: new kms.EncryptionKey(stack, 'MyKey')
+  ...
+});
+```
+* [KmsManaged](https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingKMSEncryption.html) - Server-side encryption (`SSE-KMS`), like `Kms`, except with an AWS KMS Key managed by the AWS Key Management Service.
+```ts
+new glue.Table(stack, 'MyTable', {
+  encryption: glue.TableEncryption.KmsManaged
+  ...
+});
+```
+* [ClientSideKms](https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingClientSideEncryption.html#client-side-encryption-kms-managed-master-key-intro) - Client-side encryption (`CSE-KMS`) with an AWS KMS Key managed by the account owner.
+```ts
+// KMS key is created automatically
+new glue.Table(stack, 'MyTable', {
+  encryption: glue.TableEncryption.ClientSideKms
+  ...
+});
+
+// with an explicit KMS key
+new glue.Table(stack, 'MyTable', {
+  encryption: glue.TableEncryption.ClientSideKms,
+  encryptionKey: new kms.EncryptionKey(stack, 'MyKey')
+  ...
+});
+```
+
+*Note: you cannot provide a `Bucket` when creating the `Table` if you wish to use server-side encryption (`Kms`, `KmsManaged` or `S3Managed`)*.
```

Comment:
I think we should expose `BucketEncryption` as a property on a bucket, then I could infer everything from the passed in `Bucket` and this constraint would go away. Thoughts @eladb?

Reply:
Yes, but we can only do that on Bucket and not on IBucket so that might not be good enough for what you want, right?

Reply:
We export `encryptionKey` so why not `encryption`? The only time it would be a poor experience is when importing a bucket created outside the CDK, because we'd have to ask and trust the customer to provide the right value. However, is that worse than relying on the customer to also import the `encryptionKey`, as would be the case for an externally defined `Bucket` using `BucketEncryption.Kms` (SSE-KMS)?